metabase/metabase
# add percent change to timeseries tooltip

Closes https://github.com/metabase/metabase/issues/41979

### Description

Implements [Include percent change in tooltips for time series + funnel charts](https://www.notion.so/metabase/Include-percent-change-in-tooltips-for-time-series-funnel-charts-c9f38dbe0bf9457aa1e566e267521a1d). We show this row in the tooltip for all funnel charts, and for line, area, bar, and waterfall charts with a timeseries x-axis. We will not show it after a gap in the data, e.g. if the previous month is missing, do not show the change since we have nothing to calculate it against. We also don't show it all for scatter plots, since scatter plots can have unaggregated data where it's ambigious what to compare against.

### How to verify

1. Create a line/area/bar chart with timeseries x-axis (e.g. orders sum of total grouped by month).
2. Confirm that series tooltip contains the percent change row (except if there is no data on previous row).
3. Create a funnel chart.
4. Confirm series tooltip contains both retained and change rows below the divider.

### Demo

https://github.com/metabase/metabase/assets/37751258/2d24f3cb-6253-443b-adf8-1e7cbe6b7493

### Checklist
